test/interactive/docker_test.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interactive
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+func TestEnv(t *testing.T) {
+	e := make(Env)
+	err := e.PromoteFromEnv("PWD")
+	if err != nil {
+		t.Error(err)
+		fmt.Printf("The Environ is:\n%+v\n", os.Environ())
+	}
+	if e["PWD"] != os.Getenv("PWD") {
+		t.Errorf("Promotion did not occur correctly: %v;%v", e["PWD"], os.Getenv("PWD"))
```

Comment:
```suggestion
		t.Errorf("Promotion for $PWD did not occur correctly: %s != %s", e["PWD"], os.Getenv("PWD"))
```

Reply:
actually, e["PWD"] is not guaranteed either, so will make a couple changes